Assets/Scripts/System/SceneSaveSystem.cs

```diff
@@ -0,0 +1,141 @@
+
+                this.components = new List<DclComponentData>();
+                foreach (DclComponent component in entity.Components)
+                {
+                    this.components.Add(new DclComponentData(component));
+                }
+            }
+        }
+        public struct DclComponentData
+        {
+            public string nameInCode;
+            public string nameOfSlot;
+            public List<DclComponentPropertyData> properties;
+
+            public DclComponentData(DclComponent component)
+            {
+                this.nameInCode = component.NameInCode;
+                this.nameOfSlot = component.NameOfSlot;
+                this.properties = new List<DclComponentPropertyData>();
+                foreach (DclComponent.DclComponentProperty property in component.Properties)
+                {
+                    this.properties.Add(new DclComponentPropertyData(property));
+                }
+            }
+        }
+        public struct DclComponentPropertyData
+        {
+            public string name;
+            public string type;
+            public string fixedValue;
+
+            public DclComponentPropertyData(DclComponent.DclComponentProperty property)
+            {
+                this.name = property.PropertyName;
+                this.type = property.Type.ToString();
+                object value = null;
+                switch (property.Type)
+                {
+                    case DclComponent.DclComponentProperty.PropertyType.None:
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.String:
+                        value = property.GetConcrete<string>().FixedValue;
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.Int:
+                        value = property.GetConcrete<int>().FixedValue;
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.Float:
+                        value = property.GetConcrete<float>().FixedValue;
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.Boolean:
+                        value = property.GetConcrete<bool>().FixedValue;
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.Vector3:
+                        value = property.GetConcrete<Vector3>().FixedValue;
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.Quaternion:
+                        value = property.GetConcrete<Quaternion>().FixedValue;
+                        break;
+                    case DclComponent.DclComponentProperty.PropertyType.Asset:
+                        break;
+                    default:
+                        break;
+                }
+                this.fixedValue = JsonConvert.SerializeObject(value);
```

Comment:
Hier wird das object als json serialisiert und dann als String in die entity json geschrieben. Die Idee war aber eigentlich, dass es nicht mehr diese komischen json Verschachtelungen gibt. Am besten arbeitest du hier mit JToken.
Z.B.: 
```
public struct DclComponentPropertyData
{
    public string name;
    public string type;
    public JToken value;

    public DclComponentPropertyData(DclComponent.DclComponentProperty property)
    {
        this.name = property.PropertyName;
        this.type = property.Type.ToString();
        switch (property.Type)
        {
            case DclComponent.DclComponentProperty.PropertyType.None:
                value = null;
                break;
            case DclComponent.DclComponentProperty.PropertyType.String:
                value = JToken.FromObject(property.GetConcrete<string>().FixedValue);
                break;
            case DclComponent.DclComponentProperty.PropertyType.Int:
                value = JToken.FromObject(property.GetConcrete<int>().FixedValue);
                break;
            case DclComponent.DclComponentProperty.PropertyType.Float:
                value = JToken.FromObject(property.GetConcrete<float>().FixedValue);
                break;
            case DclComponent.DclComponentProperty.PropertyType.Boolean:
                value = JToken.FromObject(property.GetConcrete<bool>().FixedValue);
                break;
            case DclComponent.DclComponentProperty.PropertyType.Vector3:
                value = JToken.FromObject(property.GetConcrete<Vector3>().FixedValue);
                break;
            case DclComponent.DclComponentProperty.PropertyType.Quaternion:
                value = JToken.FromObject(property.GetConcrete<Quaternion>().FixedValue);
                break;
            case DclComponent.DclComponentProperty.PropertyType.Asset:
                value = null;
                break;
            default:
                value = null;
                break;
        }
    }
}
```

Reply:
Ja stimmt, das war so nicht besonders sauber. Jetzt werden die Properties in folgendes Format (ohne die String Verschachtelung) serialisiert:
```
{
  "name": "rotation",
  "type": "Quaternion",
  "fixedValue": {
    "x": 0.0,
    "y": 0.0,
    "z": 0.696539164,
    "w": -0.717518747
  }
}
```
Das sollte jetzt passen, oder?